packages/kbn-openapi-bundler/src/openapi_bundler.test.ts

```diff
@@ -6,125 +6,235 @@
     it('skips nodes with x-internal property', async () => {
-      await bundleFolder('skip_internal');
-      await expectBundleToMatchFile('skip_internal', 'expected.yaml');
+      const folder = join('skip', 'skip_internal');
+
+      await bundleFolder(folder);
+      await expectBundleToMatchFile(DEFAULT_BUNDLED_FILE_PATH, join(folder, 'expected.yaml'));
     });
 
     it('skips endpoints starting with /internal', async () => {
-      await bundleFolder('skip_internal_endpoint');
-      await expectBundleToMatchFile('skip_internal_endpoint', 'expected.yaml');
+      const folder = join('skip', 'skip_internal_endpoint');
+
+      await bundleFolder(folder);
+      await expectBundleToMatchFile(DEFAULT_BUNDLED_FILE_PATH, join(folder, 'expected.yaml'));
+    });
+  });
+
+  describe('remove props', () => {
+    it('removes "x-codegen-enabled" property', async () => {
+      await bundleFolder('remove_props');
+      await expectBundleToMatchFile(
+        DEFAULT_BUNDLED_FILE_PATH,
+        join('remove_props', 'expected.yaml')
+      );
+    });
+  });
+
+  describe('reduce allOf items', () => {
+    it('flatten folded allOfs', async () => {
+      const folder = join('reduce_all_of', 'flatten_folded_all_of_items');
+
+      await bundleFolder(folder);
+      await expectBundleToMatchFile(DEFAULT_BUNDLED_FILE_PATH, join(folder, 'expected.yaml'));
+    });
+
+    it('unfolds single allOf item', async () => {
+      const folder = join('reduce_all_of', 'unfold_single_all_of_item');
+
+      await bundleFolder(folder);
+      await expectBundleToMatchFile(DEFAULT_BUNDLED_FILE_PATH, join(folder, 'expected.yaml'));
+    });
+
+    it('merges non conflicting allOf object schema items', async () => {
+      const folder = join('reduce_all_of', 'merge_non_conflicting_all_of_items');
+
+      await bundleFolder(folder);
+      await expectBundleToMatchFile(DEFAULT_BUNDLED_FILE_PATH, join(folder, 'expected.yaml'));
+    });
+
+    it('DOES NOT merge conflicting incompatible allOf object schema items', async () => {
+      const folder = join('reduce_all_of', 'do_not_merge_conflicting_all_of_items');
+
+      await bundleFolder(folder);
+      await expectBundleToMatchFile(DEFAULT_BUNDLED_FILE_PATH, join(folder, 'expected.yaml'));
+    });
+
+    it('merges allOf object schema items with inlined references', async () => {
+      const folder = join('reduce_all_of', 'merge_all_of_items_with_inlined_refs');
+
+      await bundleFolder(folder);
+      await expectBundleToMatchFile(DEFAULT_BUNDLED_FILE_PATH, join(folder, 'expected.yaml'));
+    });
+
+    it('merges allOf object schema items inlined in different document branches with extra field', async () => {
+      const folder = join(
+        'reduce_all_of',
+        'merge_all_of_items_with_inlined_refs_in_different_document_branches'
+      );
+
+      await bundleFolder(folder);
+      await expectBundleToMatchFile(DEFAULT_BUNDLED_FILE_PATH, join(folder, 'expected.yaml'));
```

Comment:
Getting to the last test in this file, I started understanding that this snapshot-like testing requires quite a lot of concentration and effort to read through and understand what is being covered. For each test, you need to read the description here, then navigate to the test folder, locate the source files and the expected YAML, then open them side by side, and compare their content to see if the input matches the expected output. This process is pretty error-prone, and as we've seen above, snapshots just captured invalid output, leaving it up to humans to verify its correctness.

I wonder what options we have to improve this experience. Maybe we could inline some of the test cases that are simple enough so the input and output are easily visible and reduce the boilerplate in the test files? What else could we do?